geth/txqueue/txqueue_manager.go

```diff
@@ -223,21 +223,24 @@ func (m *Manager) completeTransaction(queuedTx *common.QueuedTx, selectedAccount
 	if args.To != nil {
 		toAddr = *args.To
 	}
-	ctx, cancel = context.WithTimeout(context.Background(), defaultTimeout)
-	defer cancel()
-	gas, err := m.ethTxClient.EstimateGas(ctx, ethereum.CallMsg{
-		From:     args.From,
-		To:       args.To,
-		GasPrice: gasPrice,
-		Value:    value,
-		Data:     data,
-	})
-	if err != nil {
-		return emptyHash, err
-	}
-	if gas.Cmp(big.NewInt(defaultGas)) == -1 {
-		log.Info("default gas will be used. estimated gas", gas, "is lower than", defaultGas)
-		gas = big.NewInt(defaultGas)
+	gas := (*big.Int)(args.Gas)
+	if args.Gas == nil {
```

Comment:
I think it's good. The other way is to pass existing `gas` value to `EstimateGas` arguments and it will take it as upper bound, however, a lower value might be returned.

Skipping `EstimateGas` call if gas is defined should be better, though.

Reply:
also the idea was to keep EthTxClient compatible with https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L36 , it is not of high priority of course